botensure.go

```diff
@@ -0,0 +1,172 @@
+package pluginapi
+
+import (
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/blang/semver"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+	"github.com/mattermost/mattermost-server/v5/utils"
+	"github.com/pkg/errors"
+)
+
+type ensureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*ensureBotOptions)
+
+func (b *BotService) ensureServerVersion(required string) error {
+	serverVersion := b.api.GetServerVersion()
+	currentVersion := semver.MustParse(serverVersion)
+	requiredVersion := semver.MustParse(required)
+
+	if currentVersion.LT(requiredVersion) {
+		return errors.Errorf("incompatible server version for plugin, minimum required version: %s, current version: %s", required, serverVersion)
+	}
+	return nil
+}
+
+// EnsureBot either returns an existing bot user matching the given bot, or creates a bot user from the given bot.
+// A profile image or icon image may be optionally passed in to be set for the existing or newly created bot.
+// Returns the id of the resulting bot.
+//
+// Minimum server version: 5.10
+func (b *BotService) EnsureBot(bot *model.Bot, options ...EnsureBotOption) (retBotID string, retErr error) {
+	err := b.ensureServerVersion("5.10.0")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to ensure bot")
+	}
+
+	// Default options
+	o := &ensureBotOptions{
+		ProfileImagePath: "",
+		IconImagePath:    "",
+	}
```

Comment:
Probably from the original code, but is this redundant? Could we just have:
```suggestion
	o := &ensureBotOptions{}
```